### PR TITLE
BUG 1854775: Improve the MTU set/reset logic

### PR DESCRIPTION
**Pull request by pliurh**
1. When reseting PF MTU, the daemon reset it to actual initial value.
2. When setting MTU, the daemon always set the MTU of VF to the desired value.
3. Set the VF MTU to PF's if MTU is not provided by users.

### CODE REVIEW

File: pkg/daemon/writer.go [hunk 1]
```diff
@@ -33,15 +40,16 @@ func NewNodeStateStatusWriter(c snclientset.Interface, n string, f func()) *Node
...
 	glog.V(0).Infof("Run(): start writer")
 	msg := Message{}
 	if runonce {
 		glog.V(0).Info("Run(): once")
 		if err := writer.pollNicStatus(); err != nil {
 			glog.Errorf("Run(): first poll failed: %v", err)
 		}
-		writer.setNodeStateStatus(msg)
+		ns, _ := writer.setNodeStateStatus(msg)
+		writer.writeCheckpointFile(ns, destDir)
```
[the remark this answers is not in the document]
Comment (pliurh) [reply pkg/daemon/writer.go:52]: I am not sure whether we should do it. Imaging the upgrade scenario, some PFs have modified. The old pod is terminated then the new pod will be spawned. If we remove the file when the old pod is terminated, the new pod cannot get the correct initial state of those PFs.

File: pkg/utils/utils.go [hunk 2]
```diff
@@ -136,9 +134,19 @@ func SyncNodeState(newState *sriovnetworkv1.SriovNetworkNodeState) error {
...
+	mtu := ifaceStatus.Mtu
+	if iface.Mtu > 0 {
+		mtu = iface.Mtu
+		if mtu != ifaceStatus.Mtu {
+			glog.V(2).Infof("needUpdate(): MTU needs update, desired=%d, current=%d", mtu, ifaceStatus.Mtu)
+			return true
+		}
+	}
+	for _, vf := range ifaceStatus.VFs {
+		if vf.Mtu != mtu && vf.Driver != "vfio-pci" {
```
[the remark this answers is not in the document]
Comment (pliurh) [reply pkg/utils/utils.go]: Done

File: pkg/daemon/writer.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (zshi-redhat) [reply pkg/daemon/writer.go:52]: If PF is modified after SR-IOV Operator is deployed, should we restore the modified PF value or the original initial value?

**Pull request merged**